pyscal/factory.py

```diff
@@ -1044,6 +1055,46 @@ def filter_nan_from_dict(params):
     return cleaned_params
 
 
+def infer_tabular_file_format(filename):
```

Comment:
You could also look at using the inference code from xlrd to make this more reliable and with less overhead:
https://xlrd.readthedocs.io/en/latest/api.html#xlrd.inspect_format

Reply:
Thanks for you support!

Pyscal must cater for xlrd==1.2.0 still for some time, where  the `inspect_format()` is not available, so then I guess we trial-and-error approach is good for now.